components/series/stackedBar.js

```diff
@@ -6,77 +6,39 @@
         var stackedBar = function(selection) {
             var container;
 
-            // takes an object with values associated with each property, and
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
-
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
                 container = d3.select(this);
 
-                var keyFunction = function(d) {
-                    return d[stackedBar.xValueKey.value];
-                };
-                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', data, keyFunction);
+                // If the user has already set stacked data then we don't need to re-compute it.
+                var layers = stackedBar.stackedData.value;
+                if (typeof layers === 'undefined' || layers.length === 0) {
+                    layers = stackedBar.layout.value(data);
+                }
+
+                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', layers);
 
+                stackedBar.decorate.value(g);
```

Comment:
I think decorate should be moved to the end of this `selection.each` - as it is, attributes set in decorate could be overridden by the component.

Making this change this would be consistent with the other components and any manipulations which are made in decorate will take precedence over those set by the component - which I think is the right way to do it.